samples/solidity/contracts/ERC721WithData.sol

```diff
@@ -2,9 +2,10 @@
 
 pragma solidity ^0.8.0;
 
```

Comment:
I wonder if "NoData" and "WithData" are not the best names at this point. Really "NoData" is "the minimum viable ERC implementation that can still work somewhat with FireFly" and "WithData" is "all the extra recommended stuff for full FireFly functionality".

Reply:
still prefer "WithData" since it's more specific to the difference compared to nodata. plus giving it the "firefly" suffix makes it sounds as though they are supposed to be the contract of choice for using with FireFly

Reply:
Thanks for the feedback guys, I've wrote up your comments in this issue: https://github.com/hyperledger/firefly-tokens-erc20-erc721/issues/63. Going to go ahead and merge this PR.